src/Raven.Server/Documents/Handlers/TransformerHandler.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Raven.Abstractions.Indexing;
+using Raven.Server.Json;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers
+{
+    public class TransformerHandler : DatabaseRequestHandler
+    {
+        [RavenAction("/databases/*/transformers", "PUT")]
+        public async Task Put()
+        {
+            var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            {
+                var json = await context.ReadForDiskAsync(RequestBodyStream(), name);
+                var transformerDefinition = JsonDeserialization.TransformerDefinition(json);
+                transformerDefinition.Name = name;
+
+                var transformerId = Database.TransformerStore.CreateTransformer(transformerDefinition);
+
+                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
```

Comment:
I think that it  would be better to do the extension method, then. Better clarity, and one place to put it